Add changelog entries for 11.0.2 (#6007)
citusdata/citus: CHANGELOG.md

* Open sources support for row level security

* Open sources support for `pg_dist_authinfo`, which allows storing different
  authentication options for different users, e.g. you can store
  passwords or certificates here.

* Open sources support for `pg_dist_poolinfo`, which allows using connection
  poolers in between coordinator and workers

* Open sources tracking distributed query execution times using
  citus_stat_statements (`citus.stat_statements_max`,
  `citus.stat_statements_purge_interval`,
  `citus.stat_statements_track`). This is disabled by default.

* Open sources blocking tenant_isolation

* Open sources support for `sslkey` and `sslcert` in `citus.node_conninfo`

* Introduces `citus_finish_citus_upgrade()` procedure

* Introduces `synchronous` option to `citus_disable_node()` UDF

* Introduces `citus_is_coordinator` UDF to check whether a node is the
  coordinator

* Introduces `run_command_on_coordinator` UDF

* Adds support for `CREATE/DROP/ALTER VIEW` commands

* Adds support for propagating views when syncing Citus table metadata

* Enables distributed execution from `run_command_on_*` functions

* Honors `enable_metadata_sync` in node operations

> * Parallelizes metadata syncing on node activate

Reviewer: ```suggestion
* Parallelizes metadata syncing on node activation
```

> ### citus v11.0.2 (June 15, 2022) ###

Author: > are we going to skip 10.2 -> 11-0.beta changes here?

Yes. I do not like this approach personally, but this was the leading decision in some internal discussions.

I do not think that a user that wants to update from 10.2 to 11.0 will read a changelog section on a beta version and they will miss some important points.

> ### citus v11.0.2 (June 15, 2022) ###

Author: Per our discussion, I merged all the sections of `v11.0.x` into one on a single commit.

[2f00e38](https://github.com/citusdata/citus/pull/6007/commits/2f00e3855ddb2fae6aa665b1207e15bc3fd2c32d)

This commit only moves entries around. No updates, no deletes (except for `v11.0.0` and `v11.0.1` headers)

> * Introduces `citus_finish_citus_upgrade()` procedure

Reviewer: ```suggestion
* Introduces `citus_finish_citus_upgrade()` procedure that needs to be used when ...
```

> * Introduces `synchronous` option to `citus_disable_node()` UDF

Reviewer: ```suggestion
* Introduces `synchronous` option to `citus_disable_node()` UDF that allows doing foo bar ..
```

* Introduces `citus_is_coordinator` UDF to check whether a node is the coordinator

* Introduces `run_command_on_coordinator` UDF

* Adds support for `CREATE/DROP/ALTER VIEW` commands

* Adds support for propagating views when syncing Citus table metadata

> * Converts `citus.hide_shards_from_app_name_prefixes` to `citus.show_shards_for_app_name_prefixes`

Reviewer: maybe ?

```suggestion
* Replaces `citus.hide_shards_from_app_name_prefixes` with `citus.show_shards_for_app_name_prefixes`
```

### citus v11.0.2 (June 15, 2022) ###

* Drops support for PostgreSQL 12

* Open sources enterprise features, see the rest of changelog items

* Turns metadata syncing on by default

> * Adds `citus_finalize_upgrade_to_citus11()` which is necessary to upgrade to

Reviewer 2: I think we should remove ` Adds `citus_finalize_upgrade_to_citus11()` which is necessary to upgrade to Citus 11+ from earlier versions`

It is just an internal function on 11.0

Author: How about updating the next line with some information from here?

My suggestion:
* Introduces `citus_finish_citus_upgrade()` procedure which is necessary to upgrade from earlier versions?

Reviewer 2: looks good

  Citus 11+ from earlier versions

* Introduces `citus_finish_citus_upgrade()` procedure

> * Open sources non-blocking shard moves/shard rebalancer

Reviewer 2: `Open sources blocking tenant_isolation`  -> I think we can drop `blocking` because we only have 1 type of tenant isolation ?

* Open sources support for `sslkey` and `sslcert` in `citus.node_conninfo`

> * Introduces `citus_finish_citus_upgrade()` procedure

Reviewer 2: It is hard to find the point where I can find the diff

> Replaces citus.hide_shards_from_app_name_prefixes with citus.show_shards_for_app_name_prefixes

So, we should only have `citus.show_shards_for_app_name_prefixes` because the former is from the beta.

Can you check how `hide_shards_from_app_name_prefixes`  was added, and use it with `show_shards_for_app_name_prefixes` ?

Author: I see that #5567 introduced `hide_shards_from_app_name_prefixes`

The changelog entry for that was `Hides shards based on application_name prefix`.

I guess I can drop the entry for rename, and keep the entry above as is. The name of the GUC changed, but the purpose is still the same.

* Fixes schema name qualification for `ALTER/DROP STATISTICS`

> * Improves nested execution checks and adds GUC
>   `citus.allow_nested_distributed_execution`

Reviewer: maybe ?

```suggestion
* Improves nested execution checks and adds a GUC to control
  (`citus.allow_nested_distributed_execution`)
```
